diff --git a/disqus.js b/disqus.js
index 6986aa0..8057486 100644
--- a/disqus.js
+++ b/disqus.js
@@ -54,7 +54,7 @@ Drupal.behaviors.disqus = {
         // Make the AJAX call to get the Disqus comments.
         jQuery.ajax({
           type: 'GET',
-          url: 'http://' + disqus_shortname + '.disqus.com/embed.js',
+          url: '//' + disqus_shortname + '.disqus.com/embed.js',
           dataType: 'script',
           cache: false
         });
@@ -66,7 +66,7 @@ Drupal.behaviors.disqus = {
         // Make the AJAX call to get the number of comments.
         jQuery.ajax({
           type: 'GET',
-          url: 'http://' + disqus_shortname + '.disqus.com/count.js',
+          url: '//' + disqus_shortname + '.disqus.com/count.js',
           dataType: 'script',
           cache: false
         });
diff --git a/disqus.module b/disqus.module
index 2bbfa06..e3e7239 100644
--- a/disqus.module
+++ b/disqus.module
@@ -525,25 +525,25 @@ function disqus_block_view($delta = '') {
     case 'disqus_recent_comments':
       $subject = t('Recent Comments');
       $content = <<<EOT
-<div id="dsq-recentcomments" class="dsq-widget"><script type="text/javascript" src="http://disqus.com/forums/$domain/recent_comments_widget.js?num_items=$num_items&amp;excerpt_length=$excerpt_length$avatars"></script></div>
+<div id="dsq-recentcomments" class="dsq-widget"><script type="text/javascript" src="//disqus.com/forums/$domain/recent_comments_widget.js?num_items=$num_items&amp;excerpt_length=$excerpt_length$avatars"></script></div>
 EOT;
     break;
     case 'disqus_popular_threads':
       $subject = t('Popular Threads');
       $content = <<<EOT
-<div id="dsq-popthreads" class="dsq-widget"><script type="text/javascript" src="http://disqus.com/forums/$domain/popular_threads_widget.js?num_items=$num_items"></script></div>
+<div id="dsq-popthreads" class="dsq-widget"><script type="text/javascript" src="//disqus.com/forums/$domain/popular_threads_widget.js?num_items=$num_items"></script></div>
 EOT;
     break;
     case 'disqus_top_commenters':
       $subject = t('Top Commenters');
       $content = <<<EOT
-<div id="dsq-topcommenters" class="dsq-widget"><script type="text/javascript" src="http://disqus.com/forums/$domain/top_commenters_widget.js?hide_mods=$hide_mods&amp;num_items=$num_items$avatars"></script></div>
+<div id="dsq-topcommenters" class="dsq-widget"><script type="text/javascript" src="//disqus.com/forums/$domain/top_commenters_widget.js?hide_mods=$hide_mods&amp;num_items=$num_items$avatars"></script></div>
 EOT;
     break;
     case 'disqus_combination_widget':
       $subject = t('Comments');
       $content = <<<EOT
-<script type="text/javascript" src="http://disqus.com/forums/$domain/combination_widget.js?num_items=$num_items&amp;hide_mods=$hide_mods&amp;excerpt_length=$excerpt_length&amp;color=$color&amp;default_tab=$default_tab"></script>
+<script type="text/javascript" src="//disqus.com/forums/$domain/combination_widget.js?num_items=$num_items&amp;hide_mods=$hide_mods&amp;excerpt_length=$excerpt_length&amp;color=$color&amp;default_tab=$default_tab"></script>
 EOT;
     break;
     case 'disqus_comments':
