diff --git a/modules/contrib/biblio/modules/CiteProc/CSL.inc b/modules/contrib/biblio/modules/CiteProc/CSL.inc
index d761122..7016b96 100644
--- a/modules/contrib/biblio/modules/CiteProc/CSL.inc
+++ b/modules/contrib/biblio/modules/CiteProc/CSL.inc
@@ -605,7 +605,7 @@ class csl_name extends csl_format {
           }
           else {
             switch ($this->{'name-as-sort-order'}) {
-              case 'first':
+              case 'first' && $rank == 0:
               case 'all':
                 $text = $name['lastname'] . $this->sort_separator . $given;
                 break;
