diff --git a/spaces_og/plugins/space_og.inc b/spaces_og/plugins/space_og.inc
index 407025f..2244511 100644
--- a/spaces_og/plugins/space_og.inc
+++ b/spaces_og/plugins/space_og.inc
@@ -4,35 +4,39 @@
  * Organic groups integration for Spaces.
  */
 class space_og extends space_type_purl {
-  var $group = NULL;
+  public $group = NULL;
+  public $group_type;
 
   /**
    * Constructor.
    */
-  function __construct($type, $id = NULL) {
+  function __construct($type, $id = NULL){
+    $this->group_type = 'node';
     parent::__construct($type, $id);
     // Extend spaces object with og entity.
-    $this->og = og_get_group('node', $this->id);
+    if ($this->id) {
+      $this->og = entity_load_single($this->group_type, $this->id);
+    }
   }
 
   /**
    * Override of title().
    */
-  function title() {
+  function title(){
     return $this->group->title;
   }
 
   /**
    * Override of activate().
    */
-  function activate() {
+  function activate(){
     if (parent::activate()) {
       // TODO determine what replaces this in D7.
       // og_set_group_context($this->group);
       // og_set_language($this->group);
 
       // Handle theme switching for OG
-      if (!empty($this->group->og_theme)) {
+      if (! empty($this->group->og_theme)) {
         global $custom_theme;
         $custom_theme = $this->group->og_theme;
       }
@@ -45,7 +49,7 @@ class space_og extends space_type_purl {
   /**
    * Override of load().
    */
-  function load() {
+  function load(){
     // The 0 id means a new group is being saved. Instantiate a space
     // so preset values can become active.
     if ($this->id === 0) {
@@ -60,21 +64,25 @@ class space_og extends space_type_purl {
   /**
    * Override of access_space().
    */
-  function access_space($account = NULL) {
+  function access_space($account = NULL){
     global $user;
     $account = isset($account) ? $account : $user;
 
-    return parent::access_space($account) && (
-      user_access('administer group') ||
-      $this->group->{OG_ACCESS_FIELD}[LANGUAGE_NONE][0]['value'] != 1 || // Group is public, allow access
-      og_is_member($this->og->gid, 'user', $account) || // User is group member, allow access
-      user_is_anonymous()); // User hasn't logged in -- provide entry point
+    $public = (! module_exists('og_access') || $this->group->{OG_ACCESS_FIELD}[LANGUAGE_NONE][0]['value'] != 1);
+
+    return parent::access_space($account) && (user_access('administer group') || $public ||     // Group is public, allow access
+    og_is_member($this->group_type, $this->id, 'user', $account) ||     // User is
+                                                                    // group
+                                                                    // member,
+                                                                    // allow
+                                                                    // access
+    user_is_anonymous()); // User hasn't logged in -- provide entry point
   }
 
   /**
    * Override access_admin().
    */
-  function access_admin($account = NULL) {
+  function access_admin($account = NULL){
     global $user;
     $account = isset($account) ? $account : $user;
 
@@ -82,8 +90,8 @@ class space_og extends space_type_purl {
       return TRUE;
     }
     else {
-     if (user_access('administer spaces', $account) || user_access('administer group', $account)) {
-      return TRUE;
+      if (user_access('administer spaces', $account) || user_access('administer group', $account)) {
+        return TRUE;
       }
     }
     return parent::access_admin($account);
@@ -92,7 +100,7 @@ class space_og extends space_type_purl {
   /**
    * Override of access_feature().
    */
-  function access_feature($op = 'view', $feature, $account = NULL) {
+  function access_feature($op = 'view', $feature, $account = NULL){
     $access = parent::access_feature($op, $feature, $account);
 
     global $user;
@@ -101,14 +109,16 @@ class space_og extends space_type_purl {
     // Only allow access if: user can administer OG, group is public,
     // or user is a member.
     $membership = FALSE;
-    if ($membership = og_get_group_membership($this->og->gid, 'user', $account->uid)) {
+    if ($membership = og_get_membership($this->group_type, $this->id, 'user', $account->uid)) {
       $membership = $membership->state;
     }
-    $access = $access && (user_access('administer group') || $this->group->{OG_ACCESS_FIELD}[LANGUAGE_NONE][0]['value'] == OG_CONTENT_ACCESS_PUBLIC || $membership);
+
+    $wrapper = entity_metadata_wrapper('node', $this->group);
+    $access = $access && (user_access('administer group') || (!module_exists('og_access') || empty($wrapper->{OG_ACCESS_FIELD}) || !$wrapper->{OG_ACCESS_FIELD}->value()) || $membership);
 
     // Additional check for group membership if authoring content in this group.
     if ($op === 'create') {
-      return $access && og_is_member($this->og->gid, 'user', $account);
+      return $access && (user_access('administer group') || og_is_member($this->group_type, $this->id, 'user', $account));
     }
     return $access;
   }
@@ -116,12 +126,12 @@ class space_og extends space_type_purl {
   /**
    * Override of access_user().
    */
-  function access_user($op = 'view', $account = NULL) {
+  function access_user($op = 'view', $account = NULL){
     global $user;
     $account = isset($account) ? $account : $user;
     // Test whether both user and account belong to current space.
-    $test_user = og_is_member($this->og->gid, 'user');
-    $test_account = og_is_member($this->og->gid, FALSE, $account->uid);
+    $test_user = og_is_member($this->group_type, $this->id, 'user');
+    $test_account = og_is_member($this->group_type, $this->id, FALSE, $account->uid);
     if (($test_user && $test_account) || user_access('view users outside groups')) {
       return parent::access_user($op, $account);
     }
@@ -131,46 +141,49 @@ class space_og extends space_type_purl {
   /**
    * Implementation of space->router().
    */
-  function router($op, $object = NULL, $is_active = TRUE) {
+  function router($op, $object = NULL, $is_active = TRUE){
     global $user;
     switch ($op) {
-      case 'init':
+      case 'init' :
         return;
-      case 'node':
+      case 'node' :
         $node = $object;
-        $form = !isset($node->nid) || isset($node->date) ? TRUE : FALSE;
+        $form = ! isset($node->nid) || isset($node->date) ? TRUE : FALSE;
 
         // Group node
         if (og_is_group_type('node', $node->type)) {
           // Don't make new groups from within a group space.
-          if (!isset($node->nid) && $this->active) {
+          if (! isset($node->nid) && $this->active) {
             $this->deactivate();
             return;
           }
-          // If editing or viewing an existing node, make sure we're in the correct space
-          else if (isset($node->nid) && (!$this->active || ($this->active && $this->id != $node->nid)) && $space = spaces_load('og', $node->nid)) {
+          // If editing or viewing an existing node, make sure we're in the
+          // correct space
+          else if (isset($node->nid) && (! $this->active || ($this->active && $this->id != $node->nid)) && $space = spaces_load('og', $node->nid)) {
             $space->activate();
             return;
           }
         }
         // Group enabled node, route to correct group context.
-        // TODO verify that this sufficiently replaces the og_is_omitted_type() function.
+        // TODO verify that this sufficiently replaces the og_is_omitted_type()
+        // function.
         else if (og_is_group_content_type('node', $node->type)) {
-          if (!empty($node->{OG_AUDIENCE_FIELD}[LANGUAGE_NONE])) {
-            foreach($node->{OG_AUDIENCE_FIELD}[LANGUAGE_NONE] as $item) {
-              $gids[] = $item['gid'];
+          if (! empty($node->{OG_AUDIENCE_FIELD}[LANGUAGE_NONE])) {
+            foreach ( $node->{OG_AUDIENCE_FIELD}[LANGUAGE_NONE] as $item ) {
+              $gids[] = $item['target_id'];
             }
           }
 
-          // If the node belongs to the current active group space, or we're in an allowable other space type, pass thru
-          if ($this->active && in_array($this->og->gid, $gids)) {
+          // If the node belongs to the current active group space, or we're in
+          // an allowable other space type, pass thru
+          if ($this->active && in_array($this->id, $gids)) {
             return;
           }
 
           // Otherwise route
           $gid = reset($gids);
           if ($gid) {
-            spaces_load('og', og_load($gid)->etid)->activate();
+            spaces_load('og', $gid)->activate();
           }
           return;
         }
@@ -179,15 +192,29 @@ class space_og extends space_type_purl {
   }
 
   // Spaces OG views filter
-  function views_filter(&$query, $base_table = '', $relationship = '') {
+  function views_filter(&$query, $base_table = '', $relationship = ''){
+
     switch ($base_table) {
-      case 'node':
-        $table = $query->ensure_table('field_data_group_audience', $relationship);
-        $query->add_where(0, "$table.group_audience_gid", $this->og->gid);
-        break;
-      case 'users':
+      case 'og_membership' :
+        if (!isset($query->space_og_proccessed)) {
+          //Add our view arguments to the first where group
+          $where_group = key($query->where);
+          $query->space_og_proccessed = 1;
+        }
+        else {
+          // If using multiple spaces_current_space we need to OR them together
+          $query->set_group_operator('OR');
+          $where_group = $query->set_where_group('AND');
+        }
+
+        // Ensure that the og_membership relationship was added
         $table = $query->ensure_table('og_membership', $relationship);
-        $query->add_where(0, "$table.gid", $this->og->gid);
+        if ($table) {
+
+          // Add og_membership relationship parameters
+          $query->add_where($where_group, $table . '.gid', $this->og->nid, '=');
+          $query->add_where($where_group, $table . '.group_type', $this->group_type, '=');
+        }
         break;
     }
   }
@@ -196,7 +223,7 @@ class space_og extends space_type_purl {
    * Custom function group_node_admin_access().
    * Check that the current menu node object is also a group node.
    */
-  function group_node_admin_access($node) {
+  function group_node_admin_access($node){
     if ($node && og_is_group_type('node', $node->type)) {
       return $this->admin_access();
     }
@@ -206,15 +233,16 @@ class space_og extends space_type_purl {
   /**
    * Check if the user is a group administator.
    *
-   * @param $group
-   *  Group entity.
-   * @param $account
-   *  User entity.
+   * @param $group Group
+   *          entity.
+   * @param $account User
+   *          entity.
    */
-  function og_is_group_admin($group, $account) {
-    $membership = og_get_group_membership($group->gid, 'user', $account->uid);
+  function og_is_group_admin($group, $account){
+    list($nid) = entity_extract_ids($this->group_type, $group);
+    $membership = og_get_membership('node', $nid, 'user', $account->uid);
     if (isset($membership->state) && $membership->state == OG_STATE_ACTIVE) {
-      return og_user_access($group->gid, 'manage group space', $account);
+      return og_user_access('node', $nid, 'manage group space', $account);
     }
     return FALSE;
   }
@@ -223,7 +251,7 @@ class space_og extends space_type_purl {
    * Override of excluded_paths().
    * Adds og, og/* paths.
    */
-  protected function excluded_paths() {
+  protected function excluded_paths(){
     $excluded = parent::excluded_paths();
     $excluded[] = 'og';
     $excluded[] = 'og/my';
diff --git a/spaces_og/spaces_og.info b/spaces_og/spaces_og.info
index dae5d20..ed22eff 100644
--- a/spaces_og/spaces_og.info
+++ b/spaces_og/spaces_og.info
@@ -1,6 +1,5 @@
 core = 7.x
 dependencies[] = "og"
-dependencies[] = "og_access"
 dependencies[] = "og_views"
 dependencies[] = "spaces"
 dependencies[] = "purl"
@@ -15,3 +14,10 @@ files[] = tests/spaces_og.test
 files[] = views/spaces_og.views.inc
 files[] = views/spaces_og_handler_field_og_group_nids.inc
 files[] = views/spaces_og_plugin_argument_validate_useringroups.inc
+
+; Information added by drupal.org packaging script on 2012-02-16
+version = "7.x-3.x-dev"
+core = "7.x"
+project = "spaces"
+datestamp = "1329352606"
+
diff --git a/spaces_og/spaces_og.module b/spaces_og/spaces_og.module
index 34ccdab..b0f3265 100644
--- a/spaces_og/spaces_og.module
+++ b/spaces_og/spaces_og.module
@@ -269,27 +269,6 @@ function spaces_og_node_update($node, $op = 'update') {
       if (!empty($node->spaces_preset_og)) {
         $space->controllers->variable->set('spaces_preset_og', $node->spaces_preset_og);
       }
-      // Configure OG directly from space object as node_save() may be
-      // called from outside of actual group space.
-      $selective = $space->controllers->variable->get('spaces_og_selective');
-      $node->og_selective = $selective === NULL ? OG_OPEN : $selective;
-      $register = $space->controllers->variable->get('spaces_og_register');
-      $node->og_register = $register === NULL ? TRUE : $register;
-      $directory = $space->controllers->variable->get('spaces_og_directory');
-      $node->og_directory = $directory === NULL ? TRUE : $directory;
-      $private = $space->controllers->variable->get('spaces_og_private');
-      $field_langauge = field_language('node', $node, 'group_access');
-      if ($op == 'update' && $private != $node->group_access[$field_langauge][0]['value']) {
-        // Privacy has changed, update og_public flags, and queue a rebuid.
-        db_query("UPDATE {og_access_post} og_p
-          INNER JOIN {og_ancestry} og_a ON og_p.nid = og_a.nid
-          INNER JOIN {node} n ON og_p.nid = n.nid
-          SET og_public = %d WHERE og_a.group_nid = %d", ! $private, $node->nid);
-        node_access_needs_rebuild(TRUE);
-      }
-      $node->og_private = $private === NULL ? FALSE : $private;
-      // TODO determine the D7 equivalent of og_update_group().
-      // og_update_group($node);
     }
 
     // Save PURL modifier.
@@ -607,8 +586,8 @@ function spaces_og_spaces_get_space_from_object($type, $object) {
       return spaces_load('og', $object->nid);
     }
     else if (og_is_group_content_type('node', $object->type)) {
-      if (isset($object->{OG_AUDIENCE_FIELD}[LANGUAGE_NONE][0]['gid'])) {
-        return spaces_load('og', og_load($object->{OG_AUDIENCE_FIELD}[LANGUAGE_NONE][0]['gid'])->etid);
+      if (isset($object->{OG_AUDIENCE_FIELD}[LANGUAGE_NONE][0]['target_id'])) {
+        return spaces_load('og', $object->{OG_AUDIENCE_FIELD}[LANGUAGE_NONE][0]['target_id']);
       }
     }
   }
@@ -639,7 +618,7 @@ function spaces_og_og_context_handler() {
 
   $space = spaces_get_space();
   if (isset($space->type) && $space->type == 'og') {
-    $gids[] = $space->og->gid;
+    $gids[$space->group_type][] = $space->og->nid;
   }
 
   return $gids;
diff --git a/spaces_og/views/spaces_og.views.inc b/spaces_og/views/spaces_og.views.inc
index 3074c72..a9a0568 100644
--- a/spaces_og/views/spaces_og.views.inc
+++ b/spaces_og/views/spaces_og.views.inc
@@ -3,39 +3,98 @@
 /**
  * Implements hook_views_plugins().
  */
-function spaces_og_views_plugins() {
+function spaces_og_views_plugins(){
   return array(
-    'argument validator' => array(
-      'useringroups' => array(
-        'title' => t("User in current user's groups"),
-        'handler' => 'spaces_og_plugin_argument_validate_useringroups',
-        'path' => drupal_get_path('module', 'spaces_og') . '/views',
-      ),
-    ),
+      'argument validator' => array(
+          'useringroups' => array(
+              'title' => t("User in current user's groups"),
+              'handler' => 'spaces_og_plugin_argument_validate_useringroups',
+              'path' => drupal_get_path('module', 'spaces_og') . '/views'
+          )
+      )
   );
 }
 
 /**
  * Implements hook_views_handlers().
  */
-function spaces_og_views_handlers() {
+function spaces_og_views_handlers(){
   return array(
-    'info' => array(
-      'path' => drupal_get_path('module', 'spaces_og') . '/views',
-    ),
-    'handlers' => array(
-      'spaces_og_handler_field_og_group_nids' => array(
-        'parent' => 'og_views_handler_field_og_group_nids',
+      'info' => array(
+          'path' => drupal_get_path('module', 'spaces_og') . '/views'
       ),
-    ),
+      'handlers' => array(
+          'spaces_og_handler_field_og_group_nids' => array(
+              'parent' => 'og_views_handler_field_og_group_nids'
+          )
+      )
   );
 }
 
 /**
  * Implements hook_views_data_alter().
  */
-function spaces_og_views_data_alter(&$data) {
-  if (isset($data['og_ancestry']['group_nid']['field']['handler'])) {
-    $data['og_ancestry']['group_nid']['field']['handler'] = 'spaces_og_handler_field_og_group_nids';
+function spaces_og_views_data_alter(&$data){
+  if (isset($data['og_membership']['gid']['field']['handler'])) {
+    $data['og_membership']['gid']['field']['handler'] = 'spaces_og_handler_field_og_group_nids';
+  }
+}
+
+/**
+ * Implements hook_views_pre_view().
+ */
+function spaces_og_views_pre_view(&$view, &$display_id, &$args){
+
+  // The Spaces Current Space filter is bound to the space data stub
+  // if we are in an og space we want to use the og membership relation
+  // to filter the posts
+  $space = spaces_get_space();
+
+  //Is this space of type OG and we are acting on users or nodes
+  if (is_object($space) && $space->type == 'og') {
+    //After we have determined that we are in an OG space lets make
+    //sure this display makes use of the spaces filter
+    $filter = $view->get_item($display_id, 'filter', 'current');
+
+    if ($filter) {
+
+      $table = false;
+      if (!isset($filter['relationship']) || $filter['relationship'] == 'none') {
+        $table = $view->base_table;
+      }
+      elseif (($rel = $view->get_item($display_id, 'relationship', $filter['relationship']))) {
+        //Get Relationship data
+        $rel_data = views_get_handler($rel['table'], $rel['field'], 'relationship');
+
+        //Use the base table for the relationship the current space filter is attached to
+        if (isset($rel_data->definition['base'])) {
+          $table = $rel_data->definition['base'];
+        }
+      }
+
+      if (in_array($table, array(
+        'file_managed',
+        'node',
+        'user'
+      ))) {
+
+        //Check to see if the relationship has already been added via the UI
+        if (! $view->get_item($display_id, 'relationship', 'og_membership_rel')) {
+          //If so lets add the Relationship
+          $view->add_item($display_id, 'relationship', $table, 'og_membership_rel');
+        }
+
+        //Set the 'current' space filter up to use the new og relationshipt
+        $view->set_item_option($display_id, 'filter', 'current', 'relationship', 'og_membership_rel');
+      }
+
+      $i = 1;
+      //For views that use multiple spaces filters tie the multiple spaces filters to thier respective og_membership tables
+      while($view->get_item($display_id, 'filter', 'current_'.$i) && $view->get_item($display_id, 'relationship', 'og_membership_rel_'.$i)){
+        //Set the 'current_n' space filter up to use a og relationship
+        $view->set_item_option($display_id, 'filter', 'current_'.$i, 'relationship', 'og_membership_rel_'.$i);
+        $i++;
+      }
+    }
   }
 }
